### DISCUSSION TITLE
Add option to skip building extension module

### DISCUSSION THREAD

**Discussion by binh-vu** [PyO3/maturin#1682] [Ideas]
Thanks for creating this awesome tool that I frequently use to build libraries mixed between Rust & Python.

One big pain point in my CI workflow is my build time in Rust took 20 - 30 minutes due to compiling heavy crates such as RocksDB. Now given that once I've done developing my core functionalities, I often only update the Python part of the library. But every time I push the code, the CI builds everything from scratch, which takes lots of time.

I wonder if having an option to skip building the extension module (e.g., by providing a prebuilt module) is something that makes sense to consider support in Maturin. This will allow me to build my CI to: (1) check if the Rust code hasn't been modified, then (2) download the prebuilt extension modules from PyPI.org, pack them into wheels, and publish them. This will significantly reduce the CI run time.

**Comment by messense** [comment]
IMO you can explore caching first.
